x/attribute/keeper/msg_server.go

```diff
@@ -43,6 +45,13 @@ func (k msgServer) AddAttribute(goCtx context.Context, msg *types.MsgAddAttribut
 		return nil, err
 	}
 
+	// create an expiration record if an expiration value is specified
+	if len(msg.Expiration) > 0 {
+		if err := k.setExpirationForAttribute(ctx, attrib, ownerAddr, msg.Expiration); err != nil {
+			return nil, err
+		}
+	}
+
```

Comment:
In order to use the default expiration stuff, `setExpirationForAttribute` needs to always be called. In that function, if there really isn't an expiration value skip adding the expiration.
```suggestion
	if err := k.setExpirationForAttribute(ctx, attrib, ownerAddr, msg.Expiration); err != nil {
		return nil, err
	}

```